datafeed_scripts/avinetworks/vs_status_script.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+"""
+AVI ControlScript that updates the `UP` meta of an NS1 Datafeed.
+
+Invoked by the AVI Controller when a "VS_UP" or "VS_DOWN" alert is raised
+(alert action configured as ControlScript) and updates `UP` meta accordingly.
+
+`datasource_id` and `ns1_apikey` global vars must be configured in order to
+update the datafeed.  `ns1_endpoint` can optionally be configured to the FQDN
+of a custom API endpoint if using Private DNS or DDI.
+"""
+import argparse
+import json
+import sys
+
+from ns1 import Config, NS1
+
+# ID of the datasource in NS1 to publish updates to
+datasource_id = "a53252f9e583c6708331a1daeb172e12"
+# NS1 apikey to use when publishing updates
+ns1_apikey = "qACMD09OJXBxT7XOuRs8"
+# FQDN of NS1 API endpoint.  Leave as is if using Managed DNS.
+ns1_endpoint = "api.nsone.net"
+
+
+def _init_api():
+    """
+    Inititialize NS1 SDK
+
+    :return: Configured NS1 API object
+    """
+    config = Config()
+    config.createFromAPIKey(ns1_apikey)
+    config["endpoint"] = ns1_endpoint
+    return NS1(config=config)
+
+
+def _update_datafeed(api, alert_info):
+    """
+    Update `UP` metadata via datafeed
+
+    :param ns1.NS1 api: NS1 API object
+    :param dict alert_info: alert information passed by Avi Vantage alert
+    """
+    up_status = None
+    event = alert_info["events"][0]
+    label = event["event_details"]["se_hm_vs_details"]["virtual_service"]
+
+    if event["event_id"] == "VS_UP":
+        up_status = True
+    elif event["event_id"] == "VS_DOWN":
+        up_status = False
+
+    if up_status is not None:
+        data = {label: {"up": up_status}}
+        source_api = api.datasource()
+        source_api.publish(datasource_id, data)
+
+
+def _main():
```

Comment:
with simple scripts like this, where the main func is short and clear, I like to just put the `if __name__ == '__main__'` line here, and move the imports that only pertain to running as a script (argparse and sys). Not as an optimization, but for readability and clarity. Sometimes I'll still use a main() func, but move the definition below `if __name__ == '__main__'`

wonder how you feel about that sort of thing more than strongly advocating you do that :)

Reply:
Interesting! Don't have strong feelings on this, I just hate writing different boiler plate for scripts then I do for modules (because limited mental capacity).  So I always use a `main()` func, since it works for all use cases.  I guess IMO it doesn't impact readability much, plus it seems like a pretty pythonic idiom at this point.

Reply:
yeah, putting the "only-for-cli" imports in the main clause is kinda "bad python", but I like it, especially for files where there's equal use cases in which you'd import funcs, or just run the file